### PR TITLE
Xin add GatherTree

### PR DESCRIPTION
**Pull request by XinWangIntel**
Add GatherTree op and tests.

### CODE REVIEW

**Comment by YangleiZouIntel** [comment]
@tzerrell Could you please help add @XinWangIntel to CI so his PR can automatically run irv10 tests? Thanks.

He got the following error 
 "PlaidML :plaidml:" is public and triggered without a user so it is unable to trigger builds in "IRv10" as it is private. 

I rerun the tests in buildkite and passed.

**Comment by tzerrell** [comment]
I think @XinWangIntel is now added to CI; I'll confirm tomorrow, so let me know if you notice any problems. I'll also need to defer reviewing this until tomorrow as well, sorry for the delay.

**Review by tzerrell — COMMENTED** [review]
Looks generally good! Later we may want to extend the testing to better cover variations in the later inputs & parameters, but IMO this uses the existing tests well and that's a separate project. I want one style change (mentioned inline).

File: plaidml/bridge/openvino/ops/gather_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+using edsl::Tensor;
+
+namespace PlaidMLPlugin {
+
+Tensor GatherTree(Tensor STEP_IDS, Tensor PARENT_IDX, Tensor MAX_SEQ_LEN, Tensor END_TOKEN) {
```
[the remark this answers is not in the document]
Comment (XinWangIntel) [reply plaidml/bridge/openvino/ops/gather_tree.cpp]: Done

**Review by tzerrell — APPROVED** [review]

File: plaidml/bridge/openvino/ops/gather_tree.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (tzerrell) [reply plaidml/bridge/openvino/ops/gather_tree.cpp]: Thanks!

**Review by tzerrell — APPROVED** [review]

**Pull request merged**